stablehlo/transforms/StablehloAggressiveFolder.cpp

```diff
@@ -56,6 +57,8 @@ namespace stablehlo {
 
 namespace {
 
+constexpr int64_t kFoldOpEltLimit = 65536;
```

Comment:
Based on my modest experience, I would say that fewer computations typically result in better performance at run-time. Therefore, any operations that can be folded at compile time should be folded. Perhaps it would be sensible to implement this limit as a pass option?

Reply:
I agree we need to offer _a lot_ of things as folder options, lets make this a future topic. Will make a GH issue.

I'm thinking of things like `FoldConstantSize` which may only allow us to fold splats to avoid expensive operations, or `FoldElementLimit` as mentioned here may be controversial - I have had to roll back changed internally since model canonicalization took 1min+ for folding / DCE.. I think a broader design on "optimization / folding control" is needed

Reply:
I believe the reason we have this limit is because  folding typically introduces new constants which can  contribute to an increase in model size which can be significant for large tensors.

Reply:
Created https://github.com/openxla/stablehlo/issues/2588

Resolving this thread in the meantime